src/main/start-main-application/lens-window/application-window/create-lens-window.injectable.ts

```diff
@@ -35,31 +35,45 @@ const createLensWindowInjectable = getInjectable({
   instantiate: (di) => {
     const createElectronWindowFor = di.inject(createElectronWindowForInjectable);
 
-    return (configuration: LensWindowConfiguration) => {
-      let browserWindow: LensWindow | undefined;
+    return (configuration: LensWindowConfiguration): LensWindow => {
+      let browserWindow: ElectronWindow | undefined;
 
       const createElectronWindow = createElectronWindowFor(Object.assign(
         {
           onClose: () => browserWindow = undefined,
         },
+
         configuration,
       ));
 
+      let windowIsOpening = false;
+
       return {
+        id: configuration.id,
+
         get visible() {
           return !!browserWindow;
         },
+
+        get opening() {
```

Comment:
Right you are, renamed. Also instead of "opening" there's now a distinction between "starting" and "showing" a window. Therefore this naming is `isStarting`.